app/storybook/documentation/ui-kit/provider/AccessTokenProvider.mdx

```diff
@@ -0,0 +1,91 @@
+import { Canvas, ArgTypes, Meta, Story, Source, Description, Primary, Controls, Stories } from '@storybook/blocks'
+import { prettier } from '../../../src/utils'
+import * as AccessTokenProviderStories from '../../../../../packages/ui-kit/src/components/AccessTokenProvider/AccessToken.stories'
+
+<Meta title="PROVIDERS/AccessTokenProvider" /> 
+
+# AccessTokenProvider
+
+The `AccessTokenProvider` allows you to provide a function that fetches an access token from your backend. By using this function, the provider will serve the fetched access token to all of its children components and will automatically refresh the token when it expires.
+
+This way, instead of manually providing the access token to each component like this:
+
+```jsx
+import React, { useState, useEffect } from 'react'
+
+import { Counter, TimeSeries, Leaderboard } from "@propeldata/ui-kit"  
+
+import { fetchToken } from './utils'
+
+function App() {
+  const [accessToken, setAccessToken] = useState(null)
+
+  useEffect(() => {
+    async function fetchAccessToken() {
+      const accessToken = await fetchToken()
+      setAccessToken(accessToken)
+    }
+
+    fetchAccessToken()
+  }, [fetchToken, setAccessToken])
+
+  return (  
+    <Counter query={{ accessToken }} />
+    <TimeSeries query={{ accessToken }} />
+    <Leaderboard query={{ accessToken }} />
+  );  
+}
+```
+
+You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function and it will automatically provide the access token to all of its children components:
```

Comment:
```suggestion
You can just wrap your app with the `AccessTokenProvider`, pass it the `fetchToken` function and it will automatically provide the access token to all of its child components:
```